### PR TITLE
[GEOT-5851] Fixed line gap functionality for MBStyle line layers

### PR DESCRIPTION
**Pull request by johngschulz**

### CODE REVIEW

File: modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/layer/LineMBLayer.java [hunk 1]
```diff
@@ -533,23 +533,17 @@ public boolean hasLinePattern() {
...
-                    sf.description(Text.text("line"), null), NonSI.PIXEL, gapStroke, lineOffset());
-
-            org.geotools.styling.Stroke blankStroke = sf.stroke(ff.literal(Color.WHITE), lineOpacity(),
-                    ff.subtract(ff.add(lineWidth(), lineGapWidth()), ff.literal(1)),
-                    lineJoin(), lineCap(), null, null);
-            LineSymbolizer blankLine = sf.lineSymbolizer(getId(), null,
-                    sf.description(Text.text("line"), null), NonSI.PIXEL, blankStroke, lineOffset());
-            symbolizers.add(gapOuterLine);
-            symbolizers.add(blankLine);
+        	    Double topOffset = getLineOffset().doubleValue() + (getLineGapWidth().doubleValue() / 2d) + getLineWidth().doubleValue()/2d ;
```
Comment (bradh) [inline modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/layer/LineMBLayer.java]: Possible issue with indentation here?

**Comment by johngschulz** [comment]
I addressed the indentation issue, and updated the commit to include GEOT-5851 in the title, as it appears to be a duplicate of GEOS-8346 and this should fix both.

**Comment by bradh** [comment]
Travis is failing to build the MBStyles module. You might want to take a look at that.

**Comment by johngschulz** [comment]
I've had some issues with timing out on the build locally - but it looks like I forgot to update the visual transformation tests to reflect the new changes.

**Comment by tbarsballe** [comment]
On the GEOS vs GEOT issue, I had incorrectly reported this issue against GEOS instead of GEOT.
I moved the ticket over on Friday, so any confusion there is my fault.

File: modules/unsupported/mbstyle/src/test/java/org/geotools/mbstyle/transform/StyleTransformTest.java [hunk 2]
```diff
@@ -256,6 +256,16 @@ public void testLineGapWidth() throws IOException, ParseException {
...
+        StyledLayerDescriptor sld = mbStyle.transform();
+        SLDTransformer styleTransform = new SLDTransformer();
+        String xml;
+		try {
+			xml = styleTransform.transform(sld);
+	        System.out.print(xml);
+		} catch (TransformerException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
```
Comment (tbarsballe) [inline modules/unsupported/mbstyle/src/test/java/org/geotools/mbstyle/transform/StyleTransformTest.java]: Looks like tabs are being used here - should be spaces.

Comment (johngschulz) [reply modules/unsupported/mbstyle/src/test/java/org/geotools/mbstyle/transform/StyleTransformTest.java]: I actually meant to remove that before pushing - I was just using that to debug/test the sld output.

**Comment by tbarsballe** [comment]
Looks good. One last request - please include the JIRA ticket number in your commit message. It makes navigating the git history easier.

**Comment by johngschulz** [comment]
Updated commit message with Jira ticket number.

**Comment by tbarsballe** [comment]
Backported to 18.x

**Pull request merged**